torch/csrc/autograd/variable.cpp

```diff
@@ -83,12 +83,16 @@ void Variable::backward(
   Engine::get_default_engine().execute(edges, inputs, keep_graph, create_graph);
 }
 
+bool Variable::is_same_impl_type(const at::Tensor &tensor) {
```

Comment:
can't we just make this a native function?  Then you wouldn't need to make your own parsing either.